### ISSUE TITLE
lcz cant use  sf_surface_physics=7? 

### ISSUE THREAD

**Issue by wbfaded** [wrf-model/WRF#1965]

**Comment by wbfaded** [comment]
INPUT LandUse = "MODIFIED_IGBP_MODIS_NOAH"
 LANDUSE TYPE = "MODIFIED_IGBP_MODIS_NOAH" FOUND          61  CATEGORIES           2  SEASONS WATER CATEGORY =           17  SNOW CATEGORY =           15
-------------- FATAL CALLED ---------------
FATAL CALLED FROM FILE:  <stdin>  LINE:    3258
module_physics_init: PX LSM option requires USGS, MODIS, or NLCD

**Comment by weiwangncar** [comment]
@coastwx Just wondering if you are interested in addressing the use of LCZ data issue.

**Comment by coastwx** [comment]
Noted. I see that recent LANDUSE.TBL iterations have expanded the num_land_cat for NOAH variations for this LCZ paradigm. RIght now the PX LSM init stops the initialization if num_land_cat does not fit MMINLU that are defined in PX LSM internal landuse tables (phys/module_sf_pxlsm_data.F). These tables may still work even with larger num_land_cat that includes LCZ definitions and a lot of unassigned classes. It may just be adding the condition for the num_land_cat = 61.

To be clear though, this would not use any LCZ information in the PX LSM. We have some internal efforts adding some urban processes in the PX LSM, but that is not near release at this point. Any modification for the use of MODIFIED_IGBP_MODIS_NOAH 61 class would just be compatibility with Noah LSM MODIS 20 class table. 

I'll need to test and may have time this week, but cannot give a clear time line. This may be simple couple hours of testing with most of that setting up a MODIS test case. I could test the exact domain @wbfaded is attempting to run if provided namelist.wps and namelist.input files.

**Comment by coastwx** [comment]
![image](https://github.com/wrf-model/WRF/assets/15362904/be25e06a-b3f2-48b8-aa9b-d2c1007206e1)

PX may work just fine if the condition for 61 landuse cats is added here. This is where I'll start FYI.

**Comment by coastwx** [comment]
@wbfaded
@weiwangncar 

I'm starting to work on this issue. Need some details on the namelist that was used for LCZ initiation. I created a geogrid file using 

 geog_data_res = 'default',

for the MODIS landuse. I've created a one week case using UNGRIB & METGRID. 

Next step is the wrfinput_d01 file. 

1) Are the steps above correct for the case above?
2) What namelist.wrf settings trigger LCZ?  use_wudapt_lcz            = 1? 

A namelist would be great. I'll proceed, but any more specific details on how to create this. My main confusion not being a user of LCZ is that my geogrid file has this MMINLU @ 21 categories. This works with PX LSM as far as I know. But the messages above indicate 61 categories, so suspect these are created in the Real.exe step and redefined in the wrfinput_d01 file.

                :MMINLU = "MODIFIED_IGBP_MODIS_NOAH" ;
                :NUM_LAND_CAT = 21 ;
                :ISWATER = 17 ;
                :ISLAKE = 21 ;

**Comment by coastwx** [comment]
@wbfaded @weiwangncar 

Worked a few hours on this early today. I'll need some advice on how to get 61 NUM_LAND_CAT defined. This is not clear. I did multiple tests but cannot get or find how to define 61 landuse categories. I ran real.exe with 61 in the namelist, but it obviously does not allow this if the met_em* files have 21. So this may be a geogrid or metgrid processing step I'm ignorant.

![image](https://github.com/wrf-model/WRF/assets/15362904/5f07f88c-f700-4672-bfb5-7606931ab7b6)

I'm setting this aside until I can get a version of the namelist.wps that was use for a 61 class geogrid file. Both a namelist.wps and namelist.input will be needed to address this issue.

For almost all PX LSM users I don't see a problem using the standard WPS preprocessing. I used WRFv4.5 and get the 21 class MODIS landuse with Lakes and we use this with no issues in PX.

Per prior message, I think the only issue would be adding 61 to the 20, 24, 40 and 50 conditional check of the number of landuse classes in the PX LSM initialization. PX LSM will work fine, but might need to reset the number of landuse internal in PX to loop over the PX data tables correctly.

thanks for any help.

**Comment by coastwx** [comment]
@weiwangncar Thank you for the guidance. This should put me on the right track. I should have some time this week to jump on this again. Will report back asap.

**Comment by coastwx** [comment]
Just an updated that I fixed the issue. I'll follow with a pull request. I did a test run for 1 day and see no strange issues. I ran a default/base case with MODIS 21 class scheme, but the underlying MODIS landuse  dataset is apparently different as the LANDUSEF array has differences in the 1-21 LU index range. So I could not isolate the LCZ part of this. However, differences follow the landuse fraction differences and look reasonable. We just cannot endorse it without using it and doing more detailed longer-period evaluation. But it works.

Code changes were in:
 module_physics_init.F      - Allow PX to run with 61 landuse classes
 module_sf_pxlsm.F           - Allow PX to continue with 61 classes set as MODIS
 module_sf_pxlsm_data.F  - PX LSM data table expanded for MODIS to cover the 61 total categories Urban used for 51-61 LCZ
 
 
 I'm trying to find time this week to submit a pull request so expect that asap.
